projects/question/src/main/java/org/batfish/question/jsonpath/JsonPathAssertion.java

```diff
@@ -0,0 +1,77 @@
+package org.batfish.question.jsonpath;
+
+import static org.batfish.question.jsonpath.JsonPathAssertionType.count;
+import static org.batfish.question.jsonpath.JsonPathAssertionType.equal;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import org.batfish.common.BatfishException;
+import org.batfish.common.util.BatfishObjectMapper;
+
+public class JsonPathAssertion {
+
+  private static final String PROP_TYPE = "type";
+
+  private static final String PROP_EXPECT = "expect";
+
+  private JsonPathAssertionType _assertionType;
+
+  private JsonNode _expect;
+
+  public boolean evaluate(ArrayNode suffixes) {
+    switch (getType()) {
+    case count:
+      if (_expect.isInt()) {
+        boolean match = (suffixes.size() == _expect.asInt());
+        return match;
+      } else {
+        throw new BatfishException("Expected value of assertion type count ("
+            + _expect.toString() + ") is not an integer");
+      }
+    case equal:
```

Comment:
add some Java-level unit tests... things that come to mind are verifying `evaluate` for true and false assertions, complex objects, and type mismatches.

(What happens if it's not an ArrayNode?)